### PR TITLE
push the smartcontract/builder image to ECR as well as DockerHub

### PR DESCRIPTION
**Pull request by topliceanu**
See https://www.pivotaltracker.com/story/show/174100720
This PR implements the first tasks of that ticker for the `builder` image!

### CODE REVIEW

File: .github/workflows/publish.yml [hunk 1]
```diff
@@ -8,13 +8,59 @@ on:
...
+          echo "::set-env name=AWS_ECR_ACCOUNT_URL::${accountID}.dkr.ecr.us-west-2.amazonaws.com/builder"
+      - name: docker login to ECR
+        env:
+          AWS_DEFAULT_REGION: '${{ secrets.AWS_DEFAULT_REGION }}'
+        run: |
+          aws ecr get-login-password --region "${AWS_DEFAULT_REGION}" --profile "default" | docker login --username AWS --password-stdin "${AWS_ECR_ACCOUNT_URL}"
+      - name: Retag and push docker image
+        run: |
+          docker tag "smartcontract/builder:1.0.35" "${AWS_ECR_ACCOUNT_URL}:1.0.35"
+          docker images
```
Comment (HenryNguyen5) [inline .github/workflows/publish.yml:65]: Looks like we can remove this ✂️

Comment (topliceanu) [reply .github/workflows/publish.yml:65]: Yes 🤦

File: .github/workflows/publish.yml [hunk 2]
```diff
@@ -8,13 +8,58 @@ on:
...
-        run: docker build .
-      - name: Publish to Registry
+      - name: Build and publish to Docker Hub
         uses: elgohr/Publish-Docker-Github-Action@master
         with:
           name: smartcontract/builder:1.0.35
           username: ${{ secrets.DOCKER_USERNAME }}
           password: ${{ secrets.DOCKER_PASSWORD }}
+      # Push to ECR
+      - name: install aws cli
```
Comment (HenryNguyen5) [inline .github/workflows/publish.yml:21]: Let's DRY this out via https://github.com/smartcontractkit/chainlink/pull/3212#pullrequestreview-460355030

Comment (topliceanu) [reply .github/workflows/publish.yml:21]: Agreed. I'm working on it! Thanks for the review!

**Comment by topliceanu** [comment]
@HenryNguyen5 I found a GHA which retrieves temp credentials for AWS. Wdyt?

**Pull request merged**